tests/system/robot/vscodeTests.py

```diff
@@ -1,19 +1,207 @@
+	_builtIn.should_contain(speech, "hello")
+	# jump to end of file
+	spy.emulateKeyPress("control+end")
+	speech = _NvdaLib.getSpeechAfterKey("leftArrow")
+	_builtIn.should_contain(speech, "o")
+	# open find bar and search
+	spy.emulateKeyPress("control+f")
+	spy.emulateKeyPress("h")
+	spy.emulateKeyPress("e")
+	speech = _NvdaLib.getSpeechAfterKey("l")
+	_builtIn.should_contain(speech, "1 of 1")
+	spy.emulateKeyPress("escape")
+	# open replace bar
+	speech = _NvdaLib.getSpeechAfterKey("control+h")
+	_builtIn.should_contain(speech, "Replace")
+	spy.emulateKeyPress("escape")
+	# type bracket for bracket matching test
+	spy.emulateKeyPress("control+end")
+	spy.emulateKeyPress("enter")
+	for c in "(a)":
+		spy.emulateKeyPress(c)
+	# jump to matching bracket
+	spy.emulateKeyPress("control+shift+\\")
+	speech = _NvdaLib.getSpeechAfterKey("numpad2")
+	_builtIn.should_contain(speech, "left paren")
+	# move line down
+	spy.emulateKeyPress("control+home")
+	speech = _NvdaLib.getSpeechAfterKey("alt+downArrow")
+	_builtIn.should_contain(speech, "hello")
+	# move line up
+	speech = _NvdaLib.getSpeechAfterKey("alt+upArrow")
+	_builtIn.should_contain(speech, "hello")
+	# undo
+	spy.emulateKeyPress("control+z")
+	speech = _NvdaLib.getSpeechAfterKey("NVDA+upArrow")
+	_builtIn.should_contain(speech, "hello")
+	# redo
+	spy.emulateKeyPress("control+y")
+	speech = _NvdaLib.getSpeechAfterKey("NVDA+upArrow")
+	_builtIn.should_contain(speech, "(a)")
+
+
+def extensions_panel():
+	"""ensure extensions panel is accessible."""
+	_vscode.start_vscode()
+	spy = _NvdaLib.getSpyLib()
+	speech = _NvdaLib.getSpeechAfterKey("control+shift+x")
+	_builtIn.should_contain(speech, "Extensions")
+	for c in 'publisher:"Microsoft"':
+		spy.emulateKeyPress(c)
+	spy.emulateKeyPress("enter")
+	# Tab to clear search button
+	spy.emulateKeyPress("tab")
+	# Tab to search results
+	speech = _NvdaLib.getSpeechAfterKey("tab")
+	_builtIn.should_contain(speech, "Extensions  list")
+	speech = _NvdaLib.getSpeechAfterKey("downArrow")
+	_builtIn.should_contain(speech, "Microsoft")
+	speech = _NvdaLib.getSpeechAfterKey("tab")
+	_builtIn.should_contain(speech, "Install")
+
+
+def terminal_panel():
+	"""ensure terminal panel is accessible."""
+	_vscode.start_vscode()
+	spy = _NvdaLib.getSpyLib()
+	speech = _NvdaLib.getSpeechAfterKey("control+`")
+	_builtIn.should_contain(speech, "Terminal 1")
+	for c in "echo foo":
+		spy.emulateKeyPress(c)
+	spy.emulateKeyPress("enter")
+	spy.wait_for_speech_to_finish()
+	speech = _NvdaLib.getSpeechAfterKey("alt+f2")
+	_builtIn.should_contain(speech, "Accessible View")
+	# Read top of terminal output
+	speech = _NvdaLib.getSpeechAfterKey("upArrow")
+	_builtIn.should_contain(speech, "foo")
+	spy.emulateKeyPress("escape")
+	# Create new terminal
+	speech = _NvdaLib.getSpeechAfterKey("control+shift+`")
+	_builtIn.should_contain(speech, "Terminal 2")
+	# TODO: Switch between terminals
+	speech = _NvdaLib.getSpeechAfterKey("?")
+	_builtIn.should_contain(speech, "Terminal 1")
```

Comment:
Note: this experience seems to be inaccessible. does anyone know of shortcut keys / keyboard methods for switching between terminals in vs code? (Cc @codeofdusk @hwf1324 @cary-rowen  perhaps?). I couldn't get the commands suggested by the command palette (alt/control+arrowKeys) to work. In fact, switching terminals seems totally inaccessible from the keyboard

Reply:
nevermind, mystery solved, control+pageUp/Down